### PR TITLE
Implement `get_itemsize`

### PR DESCRIPTION
**Pull request by jakirkham**
This provides the function `get_itemsize` to fast path identification of the `itemsize` some common `typestr`s. Anything else that cannot be easily identified is passed off to NumPy for determination.

Before:

```python
In [1]: import cupy
   ...: import numpy
   ...: 
   ...: from ucp._libs.utils import get_buffer_nbytes

In [2]: s = (10, 11, 12, 13)
   ...: a_g = cupy.arange(numpy.prod(s)).reshape(s)

In [3]: %timeit get_buffer_nbytes(a_g, check_min_size=None, cuda_support=True)
1.28 µs ± 9.6 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

After:

```python
In [1]: import cupy
   ...: import numpy
   ...: 
   ...: from ucp._libs.utils import get_buffer_nbytes

In [2]: s = (10, 11, 12, 13)
   ...: a_g = cupy.arange(numpy.prod(s)).reshape(s)

In [3]: %timeit get_buffer_nbytes(a_g, check_min_size=None, cuda_support=True)
1 µs ± 6.44 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

### CODE REVIEW

**Review by madsbk — APPROVED** [review]
LGTM

File: ucp/_libs/utils.pyx [hunk 1]
```diff
@@ -12,12 +12,70 @@ from cpython.memoryview cimport (
...
+}
+
+cpdef Py_ssize_t get_itemsize(str fmt) except *:
+    """
+    Get the itemsize of the format provided.
+    """
+    if fmt is None:
+        raise ValueError("Expected `str`, but got `None`")
+    elif fmt == "":
+        raise ValueError("Got unexpected empty `str`")
```
Comment (pentschev) [inline ucp/_libs/utils.pyx:65]: Why do we need to special case the empty string, wouldn't that be covered by the `ValueError` exception below?

**Comment by pentschev** [comment]
I took the liberty to fix the conflict @jakirkham , hope you don't mind.

File: ucp/_libs/utils.pyx (hunk 1 above)
Comment (jakirkham) [reply ucp/_libs/utils.pyx:65]: I'm not sure that error message would fit. An empty string would still raise even if NumPy was installed. So it's not some more complicated format (like `<M8[D]`) that we don't include in the `dict` above.

**Comment by jakirkham** [comment]
Thanks Peter! Pushed a few more minor cleanup fixes in the last commit. Should fix CI.

File: ucp/_libs/utils.pyx (hunk 1 above)
Comment (pentschev) [reply ucp/_libs/utils.pyx:65]: Fair enough, thanks for clarifying.

**Review by pentschev — APPROVED** [review]
LGTM, thanks @jakirkham !

**Comment by jakirkham** [comment]
Pulled out CI fixes in PR ( https://github.com/rapidsai/ucx-py/pull/598 ).

**Comment by jakirkham** [comment]
Just to add one other thing, simply calling `getattr` takes a fair bit of time (slightly faster in Cython itself). So this optimization puts us a hair above just calling `getattr`. Meaning there probably isn't much more juice to squeeze here.

```python
In [1]: import cupy
   ...: import numpy

In [2]: s = (10, 11, 12, 13)
   ...: a_g = cupy.arange(numpy.prod(s)).reshape(s)

In [3]: %timeit getattr(a_g, "__cuda_array_interface__", None)
820 ns ± 6.04 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
```

**Pull request merged**